pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -190,14 +190,12 @@ public void start() {
         });
         runnerThread.setUncaughtExceptionHandler(
                 (t, e) -> {
-                    new Thread(() -> {
-                        LOG.error("[{}] Error while consuming records", t.getName(), e);
-                        try {
-                            this.close();
-                        } catch (Exception ex) {
-                            LOG.error("[{}] Close kafka source error", t.getName(), e);
-                        }
-                    }, "Kafka Source Close Task Thread").start();
+                    LOG.error("[{}] Error while consuming records", t.getName(), e);
+                    try {
+                        notifyError((Exception) e);
```

Comment:
May throw ClassCastException

Reply:
Maybe throwable is not instance of `Exception`